subsys/net/lib/websocket/websocket.c

```diff
@@ -419,6 +423,52 @@ static int websocket_interal_disconnect(struct websocket_context *ctx)
 
 	NET_DBG("[%p] Disconnecting", ctx);
 
+	ret = websocket_send_msg(ctx->sock, NULL, 0, WEBSOCKET_OPCODE_CLOSE,
+				 true, true, 0);
+	if (ret < 0) {
+		NET_ERR("[%p] Failed to send close message (err %d).", ctx, ret);
+	}
+
+	fds.fd = ctx->real_sock;
+	fds.events = ZSOCK_POLLIN;
+
+	/* Arbitrary timeout value for waiting for close reply. */
+	start = k_uptime_get();
+	timeout = 2000;
```

Comment:
Perhaps document this what the value means

Reply:
I've moved the comment above just before setting the timeout, plus modified it a bit to be more precise.